File: app/src/main/java/de/ironjan/mensaupb/model/LocalizedMenu.java
package de.ironjan.mensaupb.model;

public class LocalizedMenu {
    private String date;
    private String name;
    private String description;
    private String category;
    private String categoryIdentifier;
    private String subcategory;
    private String restaurant;
    private String pricetype;
    private String image;
    private String key;
    private Double priceStudents;
    private Double priceWorkers;
    private Double priceGuests;
    private String[] allergens;
    private String[] badges;
    private Integer order_info;
    private NutritionalInfo nutritionalInfo;

    public LocalizedMenu(de.ironjan.mensaupb.api.model.Menu menu, boolean isEnglish) {
        date = menu.getDate();

        if (isEnglish) {
            name = menu.getName_en();
            description = menu.getDescription_en();
            category = menu.getCategory_en();
            subcategory = menu.getSubcategory_en();
        } else {
            name = menu.getName_de();
            description = menu.getDescription_de();
            category = menu.getCategory_de();
            subcategory = menu.getSubcategory_de();
        }
        categoryIdentifier = menu.getCategory();
        restaurant = menu.getRestaurant();
        pricetype = menu.getPricetype();
        image = menu.getImage();
        key = menu.getKey();
        priceStudents = menu.getPriceStudents();
        priceWorkers = menu.getPriceWorkers();
        priceGuests = menu.getPriceGuests();
        allergens = menu.getAllergens();
        badges = menu.getBadges();
        order_info = menu.getOrder_info();
        // FIXME nutritionalInfo = menu.getNutritionalInfo();
    }

    public String getDate() {
        return date;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public String getCategory() {
        return category;
    }

    public String getSubcategory() {
        return subcategory;
    }

    public String getRestaurant() {
        return restaurant;
    }

    public String getPricetype() {
        return pricetype;
    }

    public String getImage() {
        return image;
    }

    public String getKey() {
        return key;
    }

    public Double getPriceStudents() {
        return priceStudents;
    }

    public Double getPriceWorkers() {
        return priceWorkers;
    }

    public Double getPriceGuests() {
        return priceGuests;
    }

    public String[] getAllergens() {
        return allergens;
    }

    public String[] getBadges() {
        return badges;
    }

    public Integer getOrder_info() {
        return order_info;
    }

    public NutritionalInfo getNutritionalInfo() {
        return nutritionalInfo;
    }

    public Double getPrice() {
        return getPriceStudents();
    }

    public String getCategoryIdentifier() {
        return categoryIdentifier;
    }

    public int getSortOrder() {
        switch (getCategoryIdentifier()) {
            case "dish-default":
                return 0;
            case "soups":
                return 0;
            case "dish-grill":
                return 0;
            case "dish-pasta":
                return 0;
            case "dish-wok":
                return 0;
            case "sidedish":
                return 35;
            case "dessert":
                return 70;
            case "dessert-counter":
                return 70;
            case "happydinner":
                return 5;
            case "classics-evening":
                return 35;
            case "snacks":
                return 50;
            default:
                return 0;
        }
    }

    @Override
    public String toString() {
        return "LocalizedMenu{" +
                "date='" + date + '\'' +
                ", name='" + name + '\'' +
                ", categoryIdentifier='" + categoryIdentifier + '\'' +
                ", restaurant='" + restaurant + '\'' +
                ", pricetype='" + pricetype + '\'' +
                ", priceStudents=" + priceStudents +
                '}';
    }
}

File: app/src/main/kotlin/de/ironjan/mensaupb/api/model/LocalizedMenu.kt
package de.ironjan.mensaupb.api.model

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class LocalizedMenu(menu: Menu, isEnglish: Boolean) {
    private val date: String = menu.date
    var name: String? = null
        private set

    var description: String? = null
        private set

    private var category: String? = null

    private val categoryIdentifier: String

    var subcategory: String? = null
        private set

    val restaurant: String

    val pricetype: String

    val image: String

    val key: String

    val priceStudents: Double?

    val priceWorkers: Double?

    val priceGuests: Double?

    val allergens: Array<String>

    val badges: Array<String>

    val order_info: Int?

    val price: Double?
        get() = priceStudents

    val isWeighted: Boolean
        get() = "weighted" == pricetype

    init {

        if (isEnglish) {
            name = menu.name_en
            description = menu.description_en
            category = menu.category_en
            subcategory = menu.subcategory_en
        } else {
            name = menu.name_de
            description = menu.description_de
            category = menu.category_de
            subcategory = menu.subcategory_de
        }
        categoryIdentifier = menu.category
        restaurant = menu.restaurant
        pricetype = menu.pricetype
        image = menu.image
        key = menu.key
        priceStudents = menu.priceStudents
        priceWorkers = menu.priceWorkers
        priceGuests = menu.priceGuests
        allergens = menu.allergens
        badges = menu.badges
        order_info = menu.order_info
        // FIXME nutritionalInfo = menu.getNutritionalInfo();
    }

    fun getDate(): Date? {
        try {
            return SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH).parse(date)
        } catch (e: ParseException) {

        }

        return null
    }

    fun getCategory(): String {
        return if (category.isNullOrBlank())
            "No category"
        else
            category!!
    }

    override fun toString(): String {
        return "LocalizedMenu{" +
                "date='" + date + '\''.toString() +
                ", name='" + name + '\''.toString() +
                ", categoryIdentifier='" + categoryIdentifier + '\''.toString() +
                ", restaurant='" + restaurant + '\''.toString() +
                ", pricetype='" + pricetype + '\''.toString() +
                ", priceStudents=" + priceStudents +
                '}'.toString()
    }
}
